apps/bbsync/save.py

```diff
@@ -0,0 +1,59 @@
+from collectors.bzimport.collectors import BugzillaConnector
+from osidb.models import Flaw
+
+from .query import BugzillaQueryBuilder
+
+
+class BugzillaSaver(BugzillaConnector):
+    """
+    Bugzilla flaw bug save handler
+    flaw validity is assumed and not checked
+    """
+
+    class UnsaveableFlaw(Exception):
+        """
+        error caused by attempt to save a flaw which cannot be saved
+        either by its nature or due to the current saver capabilities
+        """
+
+        pass
+
+    def __init__(self, flaw):
+        """
+        init stuff
+        """
+        self.flaw = flaw
+
+    def save(self):
+        """
+        generic save serving as class entry point
+        which calls create or update handler to continue
+        returns an updated flaw instance (without saving)
+        """
+        return self.create() if self.flaw.bz_id is None else self.update()
+
+    def create(self):
+        """
+        create flaw in Bugilla
+        """
+        bqb = BugzillaQueryBuilder(self.flaw)
+        response = self.bz_conn.createbug(bqb.query)
```

Comment:
If `self.bz_conn.createbug(...)` fails for whatever reason, exception is raised inside or should we check the actual response? Does it include the status code, error message, etc.?

Reply:
There is not much info in the method documentation but it calls some validations which may definitely raise `BugzillaError` if the arguments are in the bad format (which should hopefully not happen after it is once correct) and there can definitely come errors from Bugzilla API either when the provided data are incorrect or there is some internal/processing error and also there can be errors like timeout I guess. Either case it would just quit the creation and float away to the caller. Right now I am not entirely sure how the caller will look and where is the correct place for addressing the errors but let it just propagate back to the API (in case of user call) might be a valid approach. To the return value it is always the created `Bug` instance so there is no need to check for status code or error message and so. I would leave it as it is for now and when we actually start calling it we will know better.